Improve User Feedback

AdrianAndersen opened the pull request:
This PR does several small things to give the users better feedback about how they have configured the application and what to do in certain situations. I'm open to changing wording/severity etc

1. Make it more obvious that they are logged out. This is done by actually typing out "Logg inn" for the button, and by providing an alert in the class card. I think it makes sense to type out "Logg inn" here, but keep the small logout button. This is because we really want them to log in, not so much logout 😛 
<img width="1512" alt="Screenshot 2023-12-20 at 13 34 49" src="https://github.com/mathiazom/rezervo-web/assets/26925695/06ed359e-c037-4eff-abe0-1aaa7b044384">
<img width="622" alt="Screenshot 2023-12-20 at 13 34 03" src="https://github.com/mathiazom/rezervo-web/assets/26925695/a8cc1794-514e-4b6c-a80e-e1181a90d62c">

------------------------

2. Indicate that they need to set up a chain user in class info.
<img width="622" alt="Screenshot 2023-12-20 at 13 36 16" src="https://github.com/mathiazom/rezervo-web/assets/26925695/7ce4149e-c7d2-4ef4-850b-1aef573afae2">

------------------------

3. Indicate to the user that booking is inactive, and how to fix it
<img width="519" alt="Screenshot 2023-12-20 at 13 38 40" src="https://github.com/mathiazom/rezervo-web/assets/26925695/6a39598c-9a2e-4e8c-881d-f658d5e37cc5">
<img width="517" alt="Screenshot 2023-12-20 at 13 38 49" src="https://github.com/mathiazom/rezervo-web/assets/26925695/5c4b66b2-7829-4117-8bf2-25a3a537d69d">
<img width="629" alt="Screenshot 2023-12-20 at 13 38 22" src="https://github.com/mathiazom/rezervo-web/assets/26925695/401085b8-fd69-4501-8b1f-5c149c741c78">  

> I think it makes sense to indicate that booking has been paused in the class info since it might be a little subtle to notice the small pause icon on the agenda icon

<img width="269" alt="Screenshot 2023-12-20 at 13 37 15" src="https://github.com/mathiazom/rezervo-web/assets/26925695/d2ef8295-3809-4a65-b02d-310fc4d0f3dc">  

> Note that I moved the pause icon to the actual agenda icon, since that makes a lot more sense

------------------------
4. Finally, I added an alert for when the agenda is outdated
<img width="514" alt="Screenshot 2023-12-20 at 13 39 20" src="https://github.com/mathiazom/rezervo-web/assets/26925695/d93077c2-7888-485d-acbe-9b0d8ae3f1a5">

I also added a small chain logo in the agenda to indicate that the agenda only applies to _the current_ chain.

mathiazom commented on src/components/configuration/ConfigBar.tsx:
```diff
@@ -135,11 +139,9 @@ function ConfigBar({
-                            <LoginIcon />
-                        </IconButton>
-                    </Tooltip>
+                    <Button startIcon={<LoginIcon />} href={"/api/auth/login"}>
```
Good suggestion to write out "Logg inn" :+1:

Is it just me, or would `endIcon` be more natural?

AdrianAndersen replied:
Yeah, I agree. Fixed in https://github.com/mathiazom/rezervo-web/pull/100/commits/bc65565f9f332aad12021cf0572ad789eefb55bc

mathiazom commented on src/components/modals/ClassInfo/ClassInfo.tsx:
```diff
@@ -354,6 +355,16 @@ export default function ClassInfo({
                 </Box>
             )}
             <Typography pt={2}>{_class.activity.description}</Typography>
+            {user === undefined && !isInThePast && (
```
Nice feedback!

Have you considered showing the alert above the class image, to make it more discoverable? Currently it's placed where the buttons would be, but a new user might not be aware of where those usually are.

AdrianAndersen replied:
It might still work to have everything at the bottom though, I feel like the info is mostly for users that have randomly been logged out and tries to figure out why they cannot add something to their agenda, not new users exploring.

AdrianAndersen replied:
What do you think about this?

AdrianAndersen replied:
Soo, I thought a bit about this: I think it makes sense to make the "you need to log in" and "you need to set up a chain user" alerts sticky. This way they are very visible, and it is obvious that you can scroll, which helps new users.

I also thought about making everything sticky, so that users more easily can find the "book now" and "add to agenda" buttons. However, I think that can be a bit daunting if they have the inactive booking alert. So I think just sticking the login/user alerts is wise. 

Implemented in: https://github.com/mathiazom/rezervo-web/pull/100/commits/8d48c8eeee449a9e521f6b21d8f4171973f90b08

Do you agree, or do you have other suggestions?

AdrianAndersen replied:
https://github.com/mathiazom/rezervo-web/assets/26925695/a372c475-3606-4be5-8e0f-31b7e74f969f

mathiazom replied:
This is mostly personal preference, but I'm not a huge fan of using sticky here (kind of reminds me of the infamous cookie-banners 🙃 )

It's also still at the bottom, so not as discoverable as having it below the image.

AdrianAndersen replied:
Sure, I think it looks fine above the image too! See latest comment for the latest design. ✅

mathiazom commented on src/components/modals/Agenda/Agenda.tsx:
```diff
@@ -60,8 +71,38 @@ export default function Agenda({
+                    }
+                >
+                    <AlertTitle>Utdatert timeplan</AlertTitle>
+                    En eller flere av timene i timeplanen din er ikke satt opp denne uka. Kontroller at timeplanen din
```
```suggestion
                    En eller flere av timene i timeplanen din går ikke denne uka. Kontroller at timeplanen din
```

mathiazom commented on src/components/modals/Agenda/AgendaClassItem.tsx:
```diff
@@ -65,7 +67,7 @@ export default function AgendaClassItem({
                     justifyContent: "space-between",
                     background:
-                        agendaClass._class === undefined
+                        agendaClass._class === undefined || !bookingActive
```
I think that using the same background for ghost and paused bookings could be confusing. I think the Alert might be sufficient, or maybe you could include the pause icon again in each AgendaClassItem.

AdrianAndersen replied:
Yeah, should be more than enough to have the alert on top I think. Fixed in
https://github.com/mathiazom/rezervo-web/pull/100/commits/f04dd26a23c49797e596c7d83c0eb97926e0a1f2

mathiazom commented on src/components/modals/Agenda/Agenda.tsx:
```diff
@@ -60,8 +71,38 @@ export default function Agenda({
+                    }
+                >
+                    <AlertTitle>Utdatert timeplan</AlertTitle>
+                    En eller flere av timene i timeplanen din er ikke satt opp denne uka. Kontroller at timeplanen din
+                    stemmer overens med ukas gruppetimer.
```
```suggestion
                    En eller flere av timene i planen din går ikke denne uka. Kontroller at planen din
                    stemmer overens med ukas gruppetimer.
```

AdrianAndersen replied:
I feel like that formulation ("går ikke") could be interpreted as "not possible this week" instead of saying that it has not been scheduled. 🤔

AdrianAndersen replied:
Ehh, I kinda agree after trying on the different texts. One should understand it based on the context, since you are in a "timeplan" window. Fixed in https://github.com/mathiazom/rezervo-web/pull/100/commits/0d923cf38a187e399439e555e661827410c2b261

mathiazom commented on src/components/modals/Agenda/Agenda.tsx:
```diff
@@ -44,11 +49,17 @@ export default function Agenda({
                 <Typography variant="h6" component="h2">
                     Min timeplan
                 </Typography>
+                <Tooltip title={chainProfile.name}>
+                    <Avatar sx={{ width: 22, height: 22 }} src={chainProfile.images.common.smallLogo ?? ""}>
+                        {chainProfile.identifier}
+                    </Avatar>
+                </Tooltip>
```
Is the extra logo really necessary? Do you feel that it's unclear what chain you have selected?

AdrianAndersen replied:
Well, now that I think about it, it is only really relevant to me, since I bounce around a lot 😝

AdrianAndersen replied:
Fixed in https://github.com/mathiazom/rezervo-web/pull/100/commits/d73728fb6a180ecda201f7a287c8fa16ee6e9e99

AdrianAndersen commented:
Updated the design now @mathiazom. Also fixed up the chain user login modal a bit. Let med know what you think 😽 🧑‍🎨 

<img width="631" alt="Screenshot 2023-12-20 at 16 56 52" src="https://github.com/mathiazom/rezervo-web/assets/26925695/e730b0aa-d7b7-4257-afed-5e3da13d2e6c">
<img width="772" alt="Screenshot 2023-12-20 at 16 57 05" src="https://github.com/mathiazom/rezervo-web/assets/26925695/337510cc-3678-4dd8-ab5e-302c489612b8">

mathiazom commented on src/components/modals/ClassInfo/ClassInfo.tsx:
```diff
@@ -332,6 +335,22 @@ export default function ClassInfo({
+            {user !== undefined && userConfig === undefined && !isInThePast && (
+                <Alert severity="info" sx={{ mt: 1 }}>
+                    <AlertTitle>
+                        Registrer ditt <b>{chain}</b>-medlemskap
```
```suggestion
                        Koble til <b>{chain}</b>-medlemskap
```

Looks good, but I think we should avoid using "Registrer" to be clear that you are not creating a new 3T membership, but rather giving us access to your account 🤠

AdrianAndersen replied:
Yeah, that could be confusing. I ended up using "koble til" in the header, and "tilknytte"/"knyte" (to rezervo) in the subtitles. Updated the above comment with new screenshots now. 

Implemented in: https://github.com/mathiazom/rezervo-web/pull/100/commits/705ba6b890a2d48123f450938b54a70929b84b6b

AdrianAndersen replied:
Edit: ended up on using "koble til" both places See https://github.com/mathiazom/rezervo-web/pull/100/commits/8d21b24ee78138d124ecbe72a5308274dcea2698 and updated screenshots above

AdrianAndersen commented:
Got a tiny idea during Smidig: To indicate whether there are any classes in the agenda from the config-bar icon. (see the video) I think this can be really useful as you can see at a glance whether you have anything on your agenda, without actually clicking into it. (previously it was disabled if you had nothing in the agenda) 

With that I think I should call this PR feature complete. 😛 Let me know if you have further change suggestions @mathiazom , or whether you want me to squash the commits (became quite a few) 

https://github.com/mathiazom/rezervo-web/assets/26925695/77699323-4371-4028-93fb-4a7cd4fe03e4

mathiazom reviewed (approved):
:100:
